api/v1beta1/kafkacluster_types.go

```diff
@@ -137,6 +137,16 @@ type RollingUpgradeConfig struct {
 	// distinct broker replicas with either offline replicas or out of sync replicas and the number of alerts triggered by
 	// alerts with 'rollingupgrade'
 	FailureThreshold int `json:"failureThreshold"`
+
+	// ConcurrentBrokerRestartsAllowed controls how many brokers can be restarted in parallel during a rolling upgrade. If
+	// it is set to a value greater than 1, the operator will restart up to that amount of brokers in parallel, if the
+	// brokers are within the same rack (as specified by "broker.rack" in broker read-only configs). Since using Kafka broker
+	// racks spreads out the replicas, we know that restarting multiple brokers in the same rack will not cause more than
+	// 1/Nth of the replicas of a topic-partition to be unavailable at the same time, where N is the number of racks used.
+	// This is a safe way to speed up the rolling upgrade. Note that for the rack distribution explained above, Cruise Control
+	// requires `com.linkedin.kafka.cruisecontrol.analyzer.goals.RackAwareDistributionGoal` to be configured.
+	// +optional
+	ConcurrentBrokerRestartsAllowed int `json:"concurrentBrokerRestartsAllowed,omitempty"`
```

Comment:
Suggestion: My first impression of seeing "ConcurrentBrokerRestartsAllowed" (before the comment and actual implementation) was that I thought this is a variable to turn on/off the feature

```suggestion
	ConcurrentBrokerRestartsCount int `json:"concurrentBrokerRestartsCount,omitempty"`
```

But it could be just me, and if so feel free to ignore this

Reply:
Good point, Marton also made it in the other PR, which I addressed by renaming to `ConcurrentBrokerRestartCountPerRack`.